src/polymer/polymer2-element-scanner.ts

```diff
@@ -65,16 +65,25 @@ class ElementVisitor implements Visitor {
     }
     const element = this._handleClass(node);
     if (element) {
-      element.className = className;
-      this._possibleElements.set(element.className, element);
+      const namespacedClassName = namespaceIdentifierName(className, node);
+      element.className = namespacedClassName;
+      // Set the element on both the namespaced & unnamespaced names so that we
+      // can detect registration by either name.
+      this._possibleElements.set(namespacedClassName, element);
+      this._possibleElements.set(className, element);
     }
   }
 
   enterClassDeclaration(node: estree.ClassDeclaration) {
     const element = this._handleClass(node);
     if (element) {
-      element.className = node.id.name;
-      this._possibleElements.set(element.className, element);
+      const className = node.id.name;
+      const namespacedClassName = namespaceIdentifierName(className, node);
+      element.className = namespacedClassName;
+      // Set the element on both the namespaced & unnamespaced names so that we
```

Comment:
Polymer's core does this in a few places, and I think it's a fair use-case that we should handle:

```js
  /**
   * @memberof Polymer
   */
  class CustomStyle extends HTMLElement { ... }

  window.Polymer.CustomStyle = CustomStyle;
  window.customElements.define('custom-style', CustomStyle);
```

By adding both `Polymer.CustomStyle` & `CustomStyle` to the `_possibleElements` map for this file, we can detect when the author uses either in `customElements.define()`. It has no affect on the className of the registered element.